.github/actions/build-dotnet/Build-DotNetSolutionOrProject.ps1

```diff
@@ -1,12 +1,13 @@
-﻿param (
+param (
     [string] $Configuration,
     [string] $SolutionOrProject,
     [string] $Verbosity,
     [string] $EnableCodeAnalysis,
     [string] $Version,
     [string] $Switches,
     [string] $ExpectedCodeAnalysisErrors,
-    [boolean] $CreateBinaryLog)
+    [boolean] $CreateBinaryLog,
+    [boolean] $WarningAsError)
```

Comment:
Use how it's grammatically correct: `$WarningsAsErrors`. The same with the workflow and action parameters.